azkv/keysource_test.go

```diff
@@ -0,0 +1,102 @@
+package azkv
+
+import (
+	"flag"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestAzureKeySourceFromUrl(t *testing.T) {
+	cases := []struct {
+		name              string
+		input             string
+		expectedFoundKeys int
+		expectedKeys      []MasterKey
+	}{
+		{
+			name:              "Single url",
+			input:             "https://test.vault.azure.net/keys/test-key/a2a690a4fcc04166b739da342a912c90",
+			expectedFoundKeys: 1,
+			expectedKeys: []MasterKey{
+				{
+					VaultURL: "https://test.vault.azure.net",
+					Name:     "test-key",
+					Version:  "a2a690a4fcc04166b739da342a912c90",
+				},
+			},
+		},
+		{
+			name:              "Multiple url",
+			input:             "https://test.vault.azure.net/keys/test-key/a2a690a4fcc04166b739da342a912c90,https://test2.vault.azure.net/keys/another-test-key/cf0021e8b743453bae758e7fbf71b60e",
+			expectedFoundKeys: 2,
+			expectedKeys: []MasterKey{
+				{
+					VaultURL: "https://test.vault.azure.net",
+					Name:     "test-key",
+					Version:  "a2a690a4fcc04166b739da342a912c90",
+				},
+				{
+					VaultURL: "https://test2.vault.azure.net",
+					Name:     "another-test-key",
+					Version:  "cf0021e8b743453bae758e7fbf71b60e",
+				},
+			},
+		},
+	}
+
+	for _, c := range cases {
+		t.Run(c.name, func(t *testing.T) {
+			keys := MasterKeysFromURLs(c.input)
+			if c.expectedFoundKeys != len(keys) {
+				t.Errorf("Unexpected number of keys returned, expected %d, got %d", c.expectedFoundKeys, len(keys))
+			}
+			for idx := range keys {
+				assert.Equal(t, c.expectedKeys[idx].VaultURL, keys[idx].VaultURL)
+				assert.Equal(t, c.expectedKeys[idx].Name, keys[idx].Name)
+				assert.Equal(t, c.expectedKeys[idx].Version, keys[idx].Version)
+			}
+		})
+	}
+}
+
+func TestKeyToMap(t *testing.T) {
+	key := MasterKey{
+		CreationDate: time.Date(2016, time.October, 31, 10, 0, 0, 0, time.UTC),
+		VaultURL:     "https://test.vault.azure.net",
+		Name:         "test-key",
+		Version:      "1",
+		EncryptedKey: "this is encrypted",
+	}
+	assert.Equal(t, map[string]interface{}{
+		"vaultUrl":   key.VaultURL,
+		"key":        key.Name,
+		"version":    key.Version,
+		"enc":        "this is encrypted",
+		"created_at": "2016-10-31T10:00:00Z",
+	}, key.ToMap())
+}
+
+var azureKeyAcceptanceTestUrl = flag.String("azure-key", "", "URL to Azure Key Vault (note that this can incur real costs!)")
```

Comment:
Setting this flag will cause a real interaction with the Azure service, including the call-costs.